internal/plugin/task.go

```diff
@@ -265,6 +303,49 @@ func (s *taskLauncherServer) StopTask(
 	return &empty.Empty{}, nil
 }
 
+func (s *taskLauncherServer) WatchSpec(
+	ctx context.Context,
+	args *empty.Empty,
+) (*pb.FuncSpec, error) {
+	if s.Impl == nil {
+		return nil, status.Errorf(codes.Unimplemented, "plugin does not implement: taskLauncher")
+	}
+
+	return funcspec.Spec(s.Impl.WatchTaskFunc(),
+		argmapper.Logger(s.Logger),
+		argmapper.ConverterFunc(s.Mappers...),
+		argmapper.Typed(s.internal()),
+		argmapper.FilterOutput(
+			argmapper.FilterType(reflect.TypeOf((*component.TaskResult)(nil))),
+		),
+	)
+}
+
+func (s *taskLauncherServer) WatchTask(
+	ctx context.Context,
+	args *pb.FuncSpec_Args,
+) (*pb.TaskWatch_Resp, error) {
+	internal := s.internal()
+	defer internal.Cleanup.Close()
+
+	result, err := callDynamicFunc2(s.Impl.WatchTaskFunc(), args.Args,
+		argmapper.ConverterFunc(s.Mappers...),
+		argmapper.Logger(s.Logger),
+		argmapper.Typed(ctx),
+		argmapper.Typed(internal),
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	ret := &pb.TaskWatch_Resp{}
+	if r, ok := result.(*component.TaskResult); ok {
+		ret.ExitCode = int32(r.ExitCode)
+	}
```

Comment:
Is there anything we can do in the `!ok` case? Should we just return an error that the implemented plugin was not properly implemented to return a `TaskResult`? Maybe we would of already caught that before now in the compiler.